sdk/src/azure/core/az_mqtt5_connection_hfsm.c

```diff
@@ -255,12 +299,8 @@ static az_result connecting(az_event_policy* me, az_event event)
       }
       else
       {
-        // TODO_L: Implement credential rotation & retry.
-        _az_RETURN_IF_FAILED(_az_hfsm_send_event(
-            (_az_hfsm*)me, (az_event){ .type = AZ_HFSM_EVENT_ERROR, .data = NULL }));
-
         _az_RETURN_IF_FAILED(
-            az_event_policy_send_inbound_event((az_event_policy*)this_policy, event));
```

Comment:
Should the error here still be passed to the application?

Reply:
+1 - left a note above around a DISCONNECT_IND.